src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp

```diff
@@ -304,6 +317,9 @@ std::map<std::pair<TypeIndex, String>, NodeToSubcolumnTransformer> node_transfor
     {
         {TypeIndex::QBit, "tupleElement"}, optimizeTupleOrVariantElement<DataTypeQBit>, /// QBit uses tupleElement for subcolumns
     },
+    {
+        {TypeIndex::QBit, "tupleElement"}, optimizeTupleOrVariantElement<DataTypeObject>,
```

Comment:
```suggestion
        {TypeIndex::Object, "tupleElement"}, optimizeTupleOrVariantElement<DataTypeObject>,
```

Additionally, should we have something similar to the following query to test the optimization:

```
SET optimize_functions_to_subcolumns = 1;

drop table if exists test;
create table test (json JSON(a UInt32), tuple Tuple(b UInt32, c Int32)) engine=Memory;
insert into test values ('{"a" : 42, "b" : "s1", "c" : [{"d" : 1}, {"d" : 2}, {"d" : 3}]}', (100, 200));

EXPLAIN SYNTAX run_query_tree_passes=1
select tupleElement(json, 'a') from test;

EXPLAIN SYNTAX run_query_tree_passes=1
select tupleElement(tuple, 1) from test;
```